pkg/kubelet/cadvisor/cadvisor_linux.go

```diff
@@ -117,6 +124,33 @@ func New(port uint, runtime string) (Interface, error) {
 	return cadvisorClient, nil
 }
 
+func generateCollectorHttpClient(cAdvisorCollectorClientCertFile, cAdvisorCollectorClientPrivateKeyFile string) (*http.Client, error) {
+	//Enable accessing insecure endpoints. We should be able to access metrics from any endpoint
+	tlsConfig := &tls.Config{
+		InsecureSkipVerify: true,
```

Comment:
> I don't understand this. If I run a site under https, its not my concern or responsibility if the client is actually verifying the certificates or not.

But we're not running a random site, we're running metrics gathering for our cluster.  If our client has no option to respect or verify who its speaking to (not even basic fingerprinting for: am I talking to the same person), I think that's a flaw in the same sense that I'd be upset if my browser had no option for verifying certs on an internal network.